pkg/translator/prometheus/normalize_label.go

```diff
@@ -4,19 +4,46 @@
 package prometheus // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheus"
 
 import (
+	"fmt"
 	"strings"
 	"unicode"
 
 	"go.opentelemetry.io/collector/featuregate"
 )
 
-var dropSanitizationGate = featuregate.GlobalRegistry().MustRegister(
+var dropSanitizationGate = registerOrLoadGate(
 	"pkg.translator.prometheus.PermissiveLabelSanitization",
 	featuregate.StageAlpha,
 	featuregate.WithRegisterDescription("Controls whether to change labels starting with '_' to 'key_'."),
 	featuregate.WithRegisterReferenceURL("https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8950"),
 )
 
+func registerOrLoadGate(id string, stage featuregate.Stage, opts ...featuregate.RegisterOption) *featuregate.Gate {
```

Comment:
```suggestion
// registerOrLoadGate is a temporary workaround for a featuregate collision.
// Remove this after https://github.com/open-telemetry/opentelemetry-collector/issues/9859 is fixed.
func registerOrLoadGate(id string, stage featuregate.Stage, opts ...featuregate.RegisterOption) *featuregate.Gate {
```

Reply:
also, consider naming this "mustRegisterOrLoadGate", since it panics.

Reply:
Thanks @dashpole! Added these changes